components/autocomplete_selector.tsx

```diff
@@ -32,24 +52,25 @@ export default class AutocompleteSelector extends React.PureComponent {
         listComponent: SuggestionList,
         listPosition: 'top',
     };
+    suggestionRef?: React.RefObject<HTMLDivElement>;
 
-    constructor(props) {
+    constructor(props: Props) {
         super(props);
 
         this.state = {
             input: '',
         };
     }
 
-    onChange = (e) => {
+    onChange = (e: State) => {
         if (!e || !e.target) {
             return;
         }
 
         this.setState({input: e.target.value});
     }
 
-    handleSelected = (selected) => {
+    handleSelected = (selected: Selected) => {
         this.setState({input: ''});
 
         if (this.props.onSelected) {
```

Comment:
It might be out of scope of this PR. However, I believe that we can remove some lines using optional chaining. 

There is no need to check for `this.props.onSelected`. You can replace it with `this.props.onSelected?.(selected);`. This will do the same check. Same this will apply for the lines below.

https://github.com/mattermost/mattermost-webapp/blob/9e91d5927ac2b2b2586a2a13e7663180663631be/components/autocomplete_selector.tsx#L81-L83

https://github.com/mattermost/mattermost-webapp/blob/9e91d5927ac2b2b2586a2a13e7663180663631be/components/autocomplete_selector.tsx#L94-L96

https://github.com/mattermost/mattermost-webapp/blob/9e91d5927ac2b2b2586a2a13e7663180663631be/components/autocomplete_selector.tsx#L102-L104

Reply:
Resolved